sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -137,28 +139,23 @@
     public Mono<AccessToken> authenticateWithPfxCertificate(String pfxCertificatePath, String pfxCertificatePassword,
                                                             TokenRequestContext request) {
         String authorityUrl = options.getAuthorityHost().replaceAll("/+$", "") + "/" + tenantId;
-        try {
+        return Mono.fromCallable(() -> {
             ConfidentialClientApplication.Builder applicationBuilder =
-                ConfidentialClientApplication.builder(clientId,
-                    ClientCredentialFactory.create(new FileInputStream(pfxCertificatePath), pfxCertificatePassword))
-                    .authority(authorityUrl);
+                    ConfidentialClientApplication.builder(clientId, ClientCredentialFactory.createFromCertificate(
+                                new FileInputStream(pfxCertificatePath), pfxCertificatePassword))
+                            .httpClient(new HttpPipelineAdapter(options.getHttpPipeline()))
+                            .authority(authorityUrl);
             if (options.getProxyOptions() != null) {
                 applicationBuilder.proxy(proxyOptionsToJavaNetProxy(options.getProxyOptions()));
             }
-            ConfidentialClientApplication application = applicationBuilder.build();
-            return Mono.fromFuture(application.acquireToken(
-                ClientCredentialParameters.builder(new HashSet<>(request.getScopes()))
-                    .build()))
-                .map(ar -> new AccessToken(ar.accessToken(), OffsetDateTime.ofInstant(ar.expiresOnDate().toInstant(),
-                    ZoneOffset.UTC)));
-        } catch (CertificateException
-            | UnrecoverableKeyException
-            | NoSuchAlgorithmException
-            | KeyStoreException
-            | NoSuchProviderException
-            | IOException e) {
-            return Mono.error(e);
-        }
+            if (options.getHttpPipeline() != null) {
+                applicationBuilder.httpClient(new HttpPipelineAdapter(options.getHttpPipeline()));
```

Comment:
we're already setting the pipeline on top, even before this check.

Reply:
This actually should be used